onnx/backend/test/case/node/cast.py

```diff
@@ -29,10 +30,40 @@ def export():  # type: () -> None
             ('DOUBLE', 'FLOAT16'),
             ('FLOAT', 'STRING'),
             ('STRING', 'FLOAT'),
+            ('FLOAT', 'BFLOAT16'),
+            ('BFLOAT16', 'FLOAT'),
         ]
 
         for from_type, to_type in test_cases:
-            if 'STRING' != from_type:
+            input_type = None
+            output_type = None
+            if 'BFLOAT16' == from_type or 'BFLOAT16' == to_type:
+                np_fp32 = np.array([u'0.47892547', u'0.48033667', u'0.49968487', u'0.81910545',
+                    u'0.47031248', u'0.816468', u'0.21087195', u'0.7229038',
+                    u'NaN', u'INF', u'+INF', u'-INF'], dtype=np.float32)
+                little_endisan = sys.byteorder == 'little'
+                np_uint16_view = np_fp32.view(dtype=np.uint16)
+                np_bfp16 = np_uint16_view[1::2] if little_endisan else np_uint16_view[0::2]
```

Comment:
So, I guess this trick works correctly for NaN, INF, -INF (since you have those values in the test-case)?

Reply:
yes i have the test data there, and it is done by the standard conversion between float and bfloat16. I didn't find the definition of NaN in bfloat16 as bfloat16 is not in IEEE, but as TF apply the same conversion for all the float32 numbers, i think it should be correct.